azurerm/internal/services/network/network_connection_monitor_resource.go

```diff
@@ -814,16 +858,22 @@ func flattenNetworkConnectionMonitorEndpoint(input *[]network.ConnectionMonitorE
 			address = *item.Address
 		}
 
+		var endpointType string
+		if item.Type != "" {
+			endpointType = string(item.Type)
+		}
+
 		var resourceId string
 		if item.ResourceID != nil {
 			resourceId = *item.ResourceID
 		}
 
 		v := map[string]interface{}{
-			"name":               name,
-			"address":            address,
-			"filter":             flattenNetworkConnectionMonitorEndpointFilter(item.Filter),
-			"virtual_machine_id": resourceId,
+			"name":        name,
+			"address":     address,
+			"resource_id": resourceId,
```

Comment:
We should set `virtual_machine_id` too whilst it's still in the schema

Reply:
Updated